### PR TITLE
Prefixing package name

### PR DESCRIPTION
**Pull request by markisadesignerd**
Eslint docs say to prefix the package name with `eslint-config-` to use it in `extends`:
See https://eslint.org/docs/developer-guide/shareable-configs#creating-a-shareable-config

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "root-eslint-plugin",
+  "name": "eslint-config-root-eslint-plugin",
```
Comment (kylekthompson) [inline package.json]: Should it maybe be `eslint-config-root` then? The `eslint-plugin` is probably redundant.

**Review by kylekthompson — COMMENTED** [review]
Couple Qs:

1. Mind removing the `package-lock.json`? We use yarn, so we already have a `yarn.lock`
1. Did you double check that the various projects using this still work once you update them to point to this new commit?

**Comment by TAGraves** [comment]
I suspect this naming convention is just so eslint config plugins will live together in the package.json file. Either way, I think the name here is irrelevant since each package has to explicitly give the name they want it to have an reference in GitHub.

I do think it makes sense to be consistent with the ecosystem. In order to do that, we should:
1. Make this change here (though probably name it `eslint-config-root` like Kyle suggested)
2. Update this repo to be named `eslint-config-root` as well
3. Update every project to use `eslint-config-root` as the name of the imported package.

Mind doing all of that?
2. Rename this

**Review by williams4027 — APPROVED** [review]
This will work, you will just want to make sure that the package.json in the following projects are updated after you merge (and use that final master commit hash): root-server, root-web, root-mobile, rooter-bot

I echo what Kyle and Tommy have said in terms of renaming the package and repo as well. ~~The caveat being that builds will break until updated since it will fail to pull the `root-eslint-plugin` in yarn when builds are ran.~~ I guess the repo should work fine due to backwards compatibility, according to Tommy.

**Comment by kylekthompson** [comment]
Might make sense to just make a new GitHub repo, push this git repo to that, change everything to point to the new one, then delete this one.

**Comment by TAGraves** [comment]
@kylekthompson We'd probably lose the tying of commits to PRs if we did that - not a big deal since there aren't a ton here, but renaming is simple and GH preserves backwards compatibility.

**Comment by kylekthompson** [comment]
> GH preserves backwards compatibility.

Oh really? Nice, didn't know that

**Comment by markisadesignerd** [comment]
> I suspect this naming convention is just so eslint config plugins will live together in the package.json file. Either way, I think the name here is irrelevant since each package has to explicitly give the name they want it to have an reference in GitHub.
> 
> I do think it makes sense to be consistent with the ecosystem. In order to do that, we should:
> 
> 1. Make this change here (though probably name it `eslint-config-root` like Kyle suggested)
> 2. Update this repo to be named `eslint-config-root` as well
> 3. Update every project to use `eslint-config-root` as the name of the imported package.
> 
> Mind doing all of that?
> 2. Rename this

It's a weird requirement, but I do get an error when installing the existing package with its name and trying to use it in extends.
e.g.:
```
module.exports =  {
  plugins: ["root-eslint-plugin"].
  extends: [
    "eslint:recommended",
    "plugin:root/recommended"
  ],
};
```

gives me this error:
```
ESLint couldn't find the plugin "eslint-plugin-root-eslint-plugin".
```

**Comment by markisadesignerd** [comment]
Now that I think about it, since this isn't just a config, it should be prefixed with `eslint-plugin-*` and we should follow this convention for the embedded config(s): https://eslint.org/docs/developer-guide/working-with-plugins#configs-in-plugins

**Comment by TAGraves** [comment]
> It's a weird requirement, but I do get an error when installing the existing package with its name and trying to use it in extends.

Isn't it designed that you extend pieces of it, not the whole thing, anyway?

**Comment by markisadesignerd** [comment]
> > It's a weird requirement, but I do get an error when installing the existing package with its name and trying to use it in extends.
> 
> Isn't it designed that you extend pieces of it, not the whole thing, anyway?

Edit. 

Talked to Blake. The issue is that the package name/repo was manually added to packages.json, not installed. I'm going to update the name to reflect the one we've been manually entering.

**Comment by markisadesignerd** [comment]
With this update doing 
```
npm/yarn install https://github.com/Root-App/root-eslint-plugin.git#branch-or-hash-or-whatever
```
behaves properly (places the package into `node_modules/eslint-plugin-root`

No need to rename the repo.

After merging in, we can bump the version, cut a release and update the other projects to point to that version (easier to read than a commit hash).

**Comment by kylekthompson** [comment]
> No need to rename the repo

I think we still probably want to do this to maintain consistency

**Comment by markisadesignerd** [comment]
> > No need to rename the repo
> 
> I think we still probably want to do this to maintain consistency

Fair enough. Before or after a merge?

**Comment by kylekthompson** [comment]
> Before or after a merge?

After is probably fine ¯\\\_(ツ)\_\/¯. Tommy mentioned that GitHub already handles the backwards compatibility, so you'll be able to just update the reference to this repo when you update the commit SHA in the various repos.

**Review by kylekthompson — APPROVED** [review]

**Comment by markisadesignerd** [comment]
Any opposition to adding a `/release` branch, tagging it with the version (I was going to bump it to 0.1.0), and then using the version tag (instead of the commit hash) to reference the package in the other repos?

**Comment by williams4027** [comment]
I'm not opposed as this is probably the most professional way to manage a distributed package. I would just recommend adding release directions in the README since we don't do this with any other root projects that I have seen. Once everything is laid out, it should be pretty straightforward for future developers new to the repo to iterate from.

**Comment by kylekthompson** [comment]
> using the version tag (instead of the commit hash) to reference the package in the other repos

Does that work without it actually being published anywhere?

**Pull request closed without merging**